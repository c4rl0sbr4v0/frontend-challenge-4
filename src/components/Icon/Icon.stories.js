import React from 'react';
import Icon from './index';

export default {
  title: 'Components/Icon',
  component: Icon,
};

export const Default = () => <Icon />;
