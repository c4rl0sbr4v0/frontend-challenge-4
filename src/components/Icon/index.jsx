import React from 'react';
import PropTypes from 'prop-types';

/**
 *  Icon: Render an Icon
 */
const Icon = ({ icon, size }) => {
  return (
    <span className={`icon ${size}`}>
      <i className={`fas fa-lg ${icon}`}></i>
    </span>
  );
};

export default Icon;

Icon.propTypes = {
  icon: PropTypes.string,
  size: PropTypes.string,
};
Icon.defaultProps = {
  size: 'is-large',
  icon: 'fa-paper-plane',
};
