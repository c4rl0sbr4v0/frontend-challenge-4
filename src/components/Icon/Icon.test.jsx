import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Icon from '.';

test('Icon Component', async () => {
  window.alert = jest.fn();
  const { container } = render(<Icon />);
  expect(container.firstChild.classList.contains('icon')).toBe(true);
  expect(container).toMatchSnapshot();
});
