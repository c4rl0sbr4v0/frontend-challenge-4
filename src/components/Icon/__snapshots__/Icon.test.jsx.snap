// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component 1`] = `
<div>
  <span
    class="icon is-large"
  >
    <i
      class="fas fa-lg fa-paper-plane"
    />
  </span>
</div>
`;
